src/tests/App.test.tsx

```diff
@@ -0,0 +1,83 @@
+import { render, fireEvent, screen, waitFor } from "@testing-library/react";
+import fetch from "jest-fetch-mock";
+import React from "react";
+import { Provider } from "react-redux";
+import { store } from "../redux/store";
+import App from "../App";
+import { RESULTS } from "./stubs";
+import { createStore } from "redux";
+import { rootReducer } from "../redux/reducers";
+import { Difficulty, fetchQuizQuestions } from "../API";
+
+describe("App", () => {
+
+
+  test("sucessful game flow", async () => {
+    render(
+      <Provider store={store}>
+        <App />
+      </Provider>
+    );
+    fetch.mockResponseOnce(JSON.stringify(RESULTS));
+
+    const startButton = screen.getByText("Start");
+
+    //Fetch test
+    expect(startButton).toBeVisible();
+    fireEvent.click(startButton);
+    expect(fetch).toHaveBeenCalledTimes(1);
+
+    //Loading state
+    expect(screen.getByText("Loading...")).toBeVisible();
+    expect(screen.getByText(/Score:/i)).toBeVisible();
+
+    //Loaded state
+    await waitFor(() => screen.getByTestId("questionCard"));
+    expect(screen.getByTestId("questionCard")).toBeVisible();
+  });
+
+
+
+  test("question card use", async () => {
+    
+    fetch.mockResponseOnce(JSON.stringify(RESULTS));
+    const result = await fetchQuizQuestions(2, Difficulty.Easy);
```

Comment:
you are using the fetch here to transform your data so it's good for store, but you really shouldn't. Generally try avoiding calling any functions from your code directly if you are not unit testing stuff.
With the second test we want to not have to mock fetch or call fetch at all. I guess you may need to make another stub with the data already transformed the way store takes it. Or even transform it directly here